gammapy/irf/tests/test_energy_dispersion.py

```diff
@@ -68,21 +73,29 @@ def test_EnergyDispersion2D():
     desired = edisp.dispersion[off_node, m_node, e_node]
     assert_allclose(actual, desired, rtol=1e-06)
 
-    # Check that values between node make sense
-    # TODO: THINK ABOUT WHAT MAKES SENSE
-    energy2 = edisp.energy[e_node + 1]
-    upper = edisp.evaluate(offset, energy, migra)
-    lower = edisp.evaluate(offset, energy2, migra)
-    e_val = (energy + energy2) / 2
-    actual = edisp.evaluate(offset, e_val, migra)
-    # assert_equal(lower > actual and actual > upper, True)
+    # Check evaluation at all nodes
+    actual = edisp.evaluate().shape
+    desired = (len(edisp.offset), len(edisp.migra), len(edisp.energy))
+    assert_equal(actual, desired)
+
+    # Get response
+    e_reco = EnergyBounds.equal_log_spacing(1 * u.GeV, 100 * u.TeV, 100)
+    response = edisp.get_response('1 deg', '1 TeV', e_reco)
+    actual = len(response)
+    desired = e_reco.nbins
+    assert_equal(actual, desired)
+
+    # Check normalization
+    actual = np.sum(response)
+    desired = 1
+    assert_allclose(actual, desired, rtol=1e-1)
```

Comment:
Short answer: I don't know. We'll have to discuss / investigate where edisp PDF normalisation is important.

I think at the moment Sherpa is applying the energy dispersion, and it's a black box / not clear if the inputs need to be normalised and with what precision, no?
Maybe the RMF format has something to say about normalisation?

In the end normalisation at the ~ 1% level is important, to get the correct number of predicted counts per reco bin for a given source spectrum, i.e. a flux amplitude that is correct at the ~ 1% level.
What's not so important is the exact shape of the dispersion, i.e. pixel binning / interpolation effects.
So my understanding is that it's OK to have low-precision edisp stored and interpolated onto the RMF matrix, and only then than that RMF matrix has to be normalised (unless Sherpa is doing that, then we have to do nothing).

This is something we can discuss with Regis on Friday.

It would be good if you can find out what Sherpa does / the RMF format says, or write some test scripts, e.g. for a simulated power-law spectrum or spectral line (e.g. modeled as a Gaussian in log(Energy) or Energy) ... can you get correct npred and flux for those if some energy resolution is involved?